feat(dcglare): added dcglare command and options

mikkelkp opened the pull request:
Allows both two-phase and three-phase matrices. 'dctotal' will be replaced by vmtx*tmtx*dmtx if transmission matrix (tmtx) is given.

chriswmackey reviewed (approved):
The use of the existing modules and classes in the package looks good to me. My only suggestion is to add a test or two. It could just be a few simple ones like [what we have for rcontrib](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/tests/rcontrib_test.py).

We should probably just have someone who knows the Radiance command look over it before we merge. @mostaphaRoudsari or @sariths , would one of you be able to give a 👍 on for the Radiance side of things.

mostaphaRoudsari commented on honeybee_radiance_command/dcglare.py:
```diff
@@ -0,0 +1,186 @@
+    Properties:
+        * options
+        * output
+        * dcdirect
+        * dctotal
```
IMO, it will be more readable to use `dc_direct` and `dc_total`.

```suggestion
        * dc_direct
        * dc_total
```

mostaphaRoudsari commented on honeybee_radiance_command/dcglare.py:
```diff
@@ -0,0 +1,186 @@
+        dctotal: File path to the total (direct and diffuse) contribution file
+            (Default: None).
+        sky_matrix: File path to the sky contribution file (Default: None).
+
```
Missing docstring for `vmtx`, `dmtx` and `tmtx`

mostaphaRoudsari commented on honeybee_radiance_command/dcglare.py:
```diff
@@ -0,0 +1,186 @@
+        self.validate(stdin_input)
+
+        command_parts = [self.command, self.options.to_radiance()]
+        command_parts += [self.dcdirect]
+        command_parts += [self.dctotal] if not self.tmtx else [self.vmtx, self.tmtx,
+                          self.dmtx]
+        command_parts += [self.sky_matrix] if self.sky_matrix else []
```
Why don't you simply append these values to the list instead of creating a new list every time?

mostaphaRoudsari replied:
I was wrong! `+=` adds the item to the existing list and this is totally fine.

mostaphaRoudsari reviewed (commented):
Hi @mikkelkp, I haven't used this command enough to be able to see any possible issues. I left two comments about naming and also implementation details.

I strongly suggest to add tests to this PR before merging it in to ensure the command is working as expected.

mikkelkp commented:
> 
> 
> Hi @mikkelkp, I haven't used this command enough to be able to see any possible issues. I left two comments about naming and also implementation details.
> 
> I strongly suggest to add tests to this PR before merging it in to ensure the command is working as expected.

I changed the naming and added missing docstrings. Also a minor change in to_radiance method. Let me know if the tests file is reasonable.

mostaphaRoudsari commented on tests/dcglare_test.py:
```diff
@@ -0,0 +1,57 @@
+    assert dcglare.options.to_radiance() == ''
+
+
+def test_assignment():
+    dcglare = Dcglare()
+    dcglare.dc_direct = 'dc_direct.mtx'
+    assert dcglare.dc_direct == 'dc_direct.mtx'
+    dcglare.dc_total = 'dc_total.mtx'
+    assert dcglare.dc_total == 'dc_total.mtx'
+    dcglare.sky_matrix = 'sky_matrix.mtx'
+    assert dcglare.sky_matrix == 'sky_matrix.mtx'
+    assert dcglare.to_radiance() == 'dcglare dc_direct.mtx dc_total.mtx sky_matrix.mtx'
+    dcglare.output = 'views.dgp'
+    assert dcglare.output == 'views.dgp'
+    assert dcglare.to_radiance() == \
+        'dcglare dc_direct.mtx dc_total.mtx sky_matrix.mtx > views.dgp'
+    dcglare.options.vf = 'views.ray'
+    assert dcglare.to_radiance() == \
+        'dcglare -vf views.ray dc_direct.mtx dc_total.mtx sky_matrix.mtx > views.dgp'
```
You may want to break this into smaller unit tests.

mostaphaRoudsari reviewed (approved):
Hi @mikkelkp! Looks good to me! Feel free to merge it in. I added one comment about breaking down the test into multiple tests but it's not critical.
